lib/ash_archival/resource/changes/archive_related.ex

```diff
@@ -67,12 +80,18 @@ defmodule AshArchival.Resource.Changes.ArchiveRelated do
       destroy_action =
         Ash.Resource.Info.primary_action!(relationship.destination, :destroy).name
 
+      arguments =
+        case AshArchival.Resource.Info.archive_archive_related_arguments(resource) do
+          {:ok, {module, options}} -> module.arguments(arguments, relationship, options)
+          _ -> %{}
+        end
+
```

Comment:
I just used the arguments here because they should be the same for every changeset. I wasn't sure if there was a clean way to pass in the whole changeset when we only get one set of arguments for all of them.